sdk/src/main/java/co/omise/android/ui/AuthorizingPaymentActivity.kt

```diff
@@ -265,8 +265,12 @@ class AuthorizingPaymentActivity : AppCompatActivity() {
         val resultIntent = Intent().apply {
             putExtra(EXTRA_AUTHORIZING_PAYMENT_RESULT, Failure(throwable))
         }
-        setResult(Activity.RESULT_OK, resultIntent)
-        finish()
+        if (arrayOf("Challenge protocol error","Challenge runtime error","3DS2 initialization failed").contains(throwable.message)) {
```

Comment:
Is there any better way of identifying the error other than this string error message?

Reply:
Currently the errors are passed to our custom exception class `OmiseException` as plain strings which is doing a compare may result in typos I would say.

Reply:
Should we create new custom exceptions for these cases?

Reply:
@danfowler  I would say something like this:
```
enum class ErrorMessage(val message: String) {
    CHALLENGE_PROTOCOL_ERROR("Challenge protocol error"),
    CHALLENGE_RUNTIME_ERROR("Challenge runtime error"),
    INITIALIZATION_FAILED("3DS2 initialization failed")
}
```

Reply:
@AnasNaouchi OK, as agreed we'll make a new ticket and separate PR.